File: simple-log/src/main/java/com/don11995/log/MapFieldInner.java
/*
 * Modified by Vladyslav Lozytskyi on 12.02.18 13:14
 * Copyright (c) 2018. All rights reserved.
 */

package com.don11995.log;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.SOURCE)
@Target(ElementType.METHOD)
public @interface MapFieldInner {
    String method();

    String[] names();
}

File: simple-log/src/main/java/com/don11995/log/MapFieldInner.kt
/*
 * Modified by Vladyslav Lozytskyi on 04.05.18 17:40
 * Copyright (c) 2018. All rights reserved.
 */

package com.don11995.log

@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.FUNCTION)
annotation class MapFieldInner(val method : String, val names : Array<String>)
